eacl/validator.go

```diff
@@ -75,15 +87,40 @@ func matchFilters(hdrSrc TypedHeaderSource, filters []Filter) int {
 				continue
 			}
 
-			// get match function
-			matchFn, ok := mMatchFns[filter.Matcher()]
-			if !ok {
-				continue
-			}
-
 			// check match
-			if !matchFn(header, &filter) {
+			switch m {
+			default: // incl. MatchNotPresent
 				continue
+			case MatchStringEqual:
+				if header.Value() != filter.Value() {
+					continue
+				}
+			case MatchStringNotEqual:
+				if header.Value() == filter.Value() {
+					continue
+				}
+			case MatchNumGT, MatchNumGE, MatchNumLT, MatchNumLE:
+				// TODO: big math simplifies coding but almost always not efficient
```

Comment:
do we really need `big.Int` at all?

Reply:
no urgent need now, but this may be relevant for user attributes

Reply:
for numbers bigger than 2>>63? do we expect them?

Reply:
from user? idk what to expect. Why not?

Reply:
imo we should have some reasonable limit. it can be inefficient and also our headers can be any size long then

Reply:
overall ur suggestion can make sense. But...

other filter values aint limited now, so why should we limit the numeric ones?

also, size restriction is more about complete table to me, not single rule or filter or value

in total, yes we _can_ restrict numeric values, but i dont see any problems with them now